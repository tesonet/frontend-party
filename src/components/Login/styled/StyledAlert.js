import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Alert } from 'reactstrap';

const StyledContainer = styled.div`
  margin-bottom: 2rem;
  min-height: 3.5rem;
  visibility: ${props => props.isOpen ? 'visible' : 'hidden'};

  & .close {
    outline: none;
  }
`;

const StyledAlert = props => (
  <StyledContainer isOpen={props.isOpen}>
    <Alert color="danger" {...props}>
      {props.children}
    </Alert>
  </StyledContainer>
);

StyledAlert.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  isOpen: PropTypes.bool,
};

export default StyledAlert;
