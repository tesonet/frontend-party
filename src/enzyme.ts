import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
