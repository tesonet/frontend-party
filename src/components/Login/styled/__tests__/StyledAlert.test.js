import React from 'react';
import { shallow } from 'enzyme';

import StyledAlert from '../StyledAlert';

describe('StyledAlert component', () => {
  const props = {
    isOpen: true,
    children: <span>Test</span>,
  };

  it('should render the <div /> element.', () => {
    const renderedComponent = shallow(<StyledAlert {...props} />).dive();

    expect(renderedComponent.type()).toEqual('div');
    expect(renderedComponent).toMatchSnapshot();
  });
});
