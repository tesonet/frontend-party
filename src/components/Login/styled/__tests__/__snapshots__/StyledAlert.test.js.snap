// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledAlert component should render the <div /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    isOpen={true}
>
    <Alert
        closeAriaLabel="Close"
        color="danger"
        isOpen={true}
        tag="div"
        transition={
            Object {
                "appear": true,
                "baseClass": "fade",
                "baseClassActive": "show",
                "enter": true,
                "exit": true,
                "in": true,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 150,
                "unmountOnExit": true,
              }
        }
    >
        <span>
            Test
        </span>
    </Alert>
</styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Alert
        closeAriaLabel="Close"
        color="danger"
        isOpen={true}
        tag="div"
        transition={
                Object {
                        "appear": true,
                        "baseClass": "fade",
                        "baseClassActive": "show",
                        "enter": true,
                        "exit": true,
                        "in": true,
                        "mountOnEnter": false,
                        "onEnter": [Function],
                        "onEntered": [Function],
                        "onEntering": [Function],
                        "onExit": [Function],
                        "onExited": [Function],
                        "onExiting": [Function],
                        "tag": "div",
                        "timeout": 150,
                        "unmountOnExit": true,
                      }
        }
>
        <span>
                Test
        </span>
</Alert>,
      "className": "sc-bdVaJa gZOevp",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <span>
          Test
</span>,
        "closeAriaLabel": "Close",
        "color": "danger",
        "isOpen": true,
        "tag": "div",
        "transition": Object {
          "appear": true,
          "baseClass": "fade",
          "baseClassActive": "show",
          "enter": true,
          "exit": true,
          "in": true,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 150,
          "unmountOnExit": true,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "span",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Alert
          closeAriaLabel="Close"
          color="danger"
          isOpen={true}
          tag="div"
          transition={
                    Object {
                              "appear": true,
                              "baseClass": "fade",
                              "baseClassActive": "show",
                              "enter": true,
                              "exit": true,
                              "in": true,
                              "mountOnEnter": false,
                              "onEnter": [Function],
                              "onEntered": [Function],
                              "onEntering": [Function],
                              "onExit": [Function],
                              "onExited": [Function],
                              "onExiting": [Function],
                              "tag": "div",
                              "timeout": 150,
                              "unmountOnExit": true,
                            }
          }
>
          <span>
                    Test
          </span>
</Alert>,
        "className": "sc-bdVaJa gZOevp",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span>
            Test
</span>,
          "closeAriaLabel": "Close",
          "color": "danger",
          "isOpen": true,
          "tag": "div",
          "transition": Object {
            "appear": true,
            "baseClass": "fade",
            "baseClassActive": "show",
            "enter": true,
            "exit": true,
            "in": true,
            "mountOnEnter": false,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 150,
            "unmountOnExit": true,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "span",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
